Glitter/Sources/main.cpp

```diff
@@ -125,33 +244,95 @@ int main(int argc, char * argv[]) {
   // position
   GLint posAttrib = glGetAttribLocation(shaderProgram, "position");
   glEnableVertexAttribArray(posAttrib);
-  glVertexAttribPointer(posAttrib, 2, GL_FLOAT, GL_FALSE, 5 * sizeof(GLfloat), 0);  // attribute location, # values, value type, normalize?, stride, offset
+  glVertexAttribPointer(posAttrib, 3, GL_FLOAT, GL_FALSE, 6 * sizeof(GLfloat), 0);  // attribute location, # values, value type, normalize?, stride, offset
   // color
   GLint colAttrib = glGetAttribLocation(shaderProgram, "color");
   glEnableVertexAttribArray(colAttrib);
-  glVertexAttribPointer(colAttrib, 3, GL_FLOAT, GL_FALSE, 5 * sizeof(GLfloat), (void*)(2 * sizeof(GLfloat)));
+  glVertexAttribPointer(colAttrib, 3, GL_FLOAT, GL_FALSE, 6 * sizeof(GLfloat), (void*)(3 * sizeof(GLfloat)));
   
   // model matrix
+  
+
+
   GLint modelTransform = glGetUniformLocation(shaderProgram, "model");
-  glm::mat4 scale_model = glm::scale(glm::mat4(1.0f), glm::vec3(0.5f,0.5f,0.0f));
+  glm::mat4 scale_model = glm::scale(glm::mat4(1.0f), glm::vec3(0.5f,0.5f,0.5f));
 	glm::mat4 model = scale_model;
   glUniformMatrix4fv(modelTransform, 1, GL_FALSE, glm::value_ptr(model));
   
   // Rendering Loop
+  
+  glDepthFunc(GL_LESS);
+  glEnable(GL_DEPTH_TEST);
+  
   while (glfwWindowShouldClose(mWindow) == false) {
     
     // Background Fill Color
     glClearColor(0.85f, 0.65f, 0.65f, 0.8f);
-    glClear(GL_COLOR_BUFFER_BIT);
+    glClearDepth(1.0f);
+    glClear(GL_COLOR_BUFFER_BIT | GL_DEPTH_BUFFER_BIT);
     
     // draw triangle
-    glDrawArrays(GL_TRIANGLES, 0, sizeof(vertices)/sizeof(vertices[0]));
-    
+    glDrawElements(GL_TRIANGLES, 12, GL_UNSIGNED_INT, 0);
+	
+
+	
+	
+	
+	
+	// Camera/View transformation
+		glm::mat4 view;
+		view = glm::lookAt(cameraPos, cameraPos + cameraFront, cameraUp);
+		// Projection 
+		glm::mat4 projection;
+		projection = glm::perspective(45.0f, (GLfloat)200 / (GLfloat)200, 0.1f, 100.0f);
+		// Get the uniform locations
+		GLint modelLoc = glGetUniformLocation(shaderProgram, "model");
+		GLint viewLoc = glGetUniformLocation(shaderProgram, "view");
+		GLint projLoc = glGetUniformLocation(shaderProgram, "projection");
+		// Pass the matrices to the shader
+		glUniformMatrix4fv(viewLoc, 1, GL_FALSE, glm::value_ptr(view));
+		glUniformMatrix4fv(projLoc, 1, GL_FALSE, glm::value_ptr(projection));
+
+
+		
+		
+		
+		
+		
+		
+		
+		
+
+		glUniformMatrix4fv(modelTransform, 1, GL_FALSE, glm::value_ptr(model));
+
+
+	
+	
+	
+	
+	
+	
+	
+	
+	
+	
+	
+	
+	
+	
+	for(int i = 0; i < 4; i++) {
+		GLfloat tmp = vertices[6*i];
+		vertices[6*i] = vertices[6*i]*cos(0.00004) + vertices[6*i+2]*sin(0.00004);
+		vertices[6*i+2] = vertices[6*i+2]*cos(0.00004) - tmp*sin(0.00004);
```

Comment:
don't manually loop over vertices. calculate a matrix once and send it to the vertex shader to be multiplied in the GPU.